src/creature.cpp

```diff
@@ -1121,24 +1121,17 @@ void Creature::onGainExperience(uint64_t gainExp, Creature* target)
 	}
 }
 
-void Creature::addSummon(Creature* creature)
-{
-	creature->setDropLoot(false);
-	creature->setLossSkill(false);
-	creature->setMaster(this);
-	creature->incrementReferenceCounter();
-	summons.push_back(creature);
-}
-
-void Creature::removeSummon(Creature* creature)
-{
-	auto cit = std::find(summons.begin(), summons.end(), creature);
-	if (cit != summons.end()) {
-		creature->setDropLoot(false);
-		creature->setLossSkill(true);
-		creature->setMaster(nullptr);
-		creature->decrementReferenceCounter();
-		summons.erase(cit);
+void Creature::setMaster(Creature* creature) {
+	master = creature;
+	if (creature) {
+		this->incrementReferenceCounter();
+		creature->summons.push_back(this);
+	} else {
+		auto cit = std::find(summons.begin(), summons.end(), this);
+		if (cit != summons.end()) {
+			this->decrementReferenceCounter();
```

Comment:
Also this may break things when using setMaster(nullptr), if its counter is decremented to 0 it will be deleted and if there is places in the source code that it doesn't expect that it will break.